app/javascript/src/Policies/util.jsx

```diff
@@ -35,10 +36,45 @@ function generateGuid (): string {
   return s4() + s4() + '-' + s4() + '-' + s4() + '-' + s4() + '-' + s4() + s4() + s4()
 }
 
+function parsePolicies (registry: RawRegistry): Array<RegistryPolicy> {
+  let policies: Array<RegistryPolicy> = []
+  for (let key in registry) {
+    registry[key].forEach(policy => policies.push(parsePolicy(key, policy)))
+  }
+  return policies
+}
+
+function parsePolicy (key: string, policy: RawPolicy): RegistryPolicy {
+  return Object.assign({}, policy,
+    {
+      name: key,
+      humanName: policy.name,
+      data: {}
+    })
+}
+
+const toJson = (val: Object): string => JSON.stringify(val, null, 2)
+
+const fromJson = (json: string) => JSON.parse(json)
+
+const safeFromJson = (json: string, callback: () => {}): Object => {
```

Comment:
I don't you need a callback in this method, since it's synchronous. Return an Error or undefined and call any further function independently.

Reply:
You are party right, but wrong about callback being only async. I'd keep the console.error and return undefined.